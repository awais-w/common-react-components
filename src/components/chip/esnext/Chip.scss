@import '@cmc/common-styles/src/scss/_variables.scss';

@mixin applyColor($color) {
  $darken-amount: 10%;
  border-color: $color;
  .cmc-icon,
  .cmc-icon-custom {
    background: $color;
  }
  .cmc-chip__label {
    color: $color;
  }
  &:hover {
    border-color: darken($color, $darken-amount);
    .cmc-icon,
    .cmc-icon-custom {
      background: darken($color, $darken-amount);
    }
    .cmc-chip__label {
      color: darken($color, $darken-amount);
    }
  }
}

.cmc-chip {
  display: inline-flex;
  align-items: center;
  height: $large;
  border: 1px solid;
  border-radius: $medium;
  cursor: pointer;

  .material-icons {
    // padding-right: $half;
    user-select: none;
  }

  .cmc-icon,
  .cmc-icon-custom {
    @include flexbox;
    height: $large;
    width: $large;
    border-radius: $medium;
    color: $color-white;
    font-size: 18px;
    margin-left: -1px;
  }

  .cmc-icon-custom::before {
    font-size: $font-size-small;
  }

  &__label {
    @include font-bold;
    font-size: $font-size-small;
    margin: 0 $base 0 $half;
  }

  &--black {
    @include applyColor($color-grey-charcoal);
  }
  &--blue {
    @include applyColor($color-cerulean);
  }
  &--red {
    @include applyColor($color-rust);
  }
  &--green {
    @include applyColor($color-green);
  }
  &--amber {
    @include applyColor($color-sainsburys);
  }
  &--grey {
    @include applyColor($color-grey-dark);
  }
}
