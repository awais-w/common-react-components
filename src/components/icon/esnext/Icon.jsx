import React from 'react';
import PropTypes from 'prop-types';
import bem from '@argos/utils/esnext/bem';
import './Icon.scss';

const Icon = (props) => {
  const { isCustom, icon, onClick, id, disabled, color } = props;
  const cn = bem({ prefix: 'cmc-', block: isCustom ? 'icon-custom' : 'icon' });
  return (
    <i
      className={`${cn(null, color, { disabled, [icon]: isCustom, clickable: onClick })}`}
      id={id || undefined}
      role={onClick ? 'button' : 'icon'}
      onClick={onClick}>
      {icon}
    </i>
  );
};

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
  isCustom: PropTypes.bool,
  onClick: PropTypes.func,
  id: PropTypes.string,
  disabled: PropTypes.bool,
  color: PropTypes.oneOf([undefined, 'white', 'red', 'yellow', 'green', 'blue', 'black', 'grey']),
};

Icon.defaultProps = {
  onClick: null,
  isCustom: false,
};

export default Icon;
