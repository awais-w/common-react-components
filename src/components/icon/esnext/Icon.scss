@import '@cmc/common-styles/src/scss/_variables.scss';

@font-face {
  font-family: 'Material Icons';
  font-style: normal;
  font-weight: 400;
  src: local('Material Icons'), local('MaterialIcons-Regular'),
    url(https://fonts.gstatic.com/s/materialicons/v42/flUhRq6tzZclQEJ-Vdg-IuiaDsNcIhQ8tQ.woff2) format('woff2');
}

@font-face {
  font-family: 'icomoon';
  src: url('fonts/icomoon.woff?y5vhvj') format('woff');
  font-style: normal;
  font-weight: 400;
}

.cmc-icon {
  font-family: 'Material Icons';
  font-weight: normal;
  font-style: normal;
  font-size: $icon-size-standard;
  line-height: 1;
  letter-spacing: normal;
  text-transform: none;
  display: inline-block;
  white-space: nowrap;
  word-wrap: normal;
  direction: ltr;
  font-feature-settings: 'liga';
  -webkit-font-smoothing: subpixel-antialiased;
}

.cmc-icon-custom {
  /* use !important to prevent issues with browser extensions that change fonts */
  font-family: 'icomoon' !important;
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;
  line-height: 1;
  font-size: 0 !important;

  /* Better Font Rendering =========== */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  &:before {
    font-size: $font-size-large;
  }
}

i.cmc-icon,
i.cmc-icon-custom {
  &.clickable {
    cursor: pointer;
  }
  &--white {
    color: $color-white;
  }
  &--red {
    color: $color-red;
  }
  &--yellow {
    color: $color-star-stroke;
  }
  &--green {
    color: $color-green;
  }
  &--blue {
    color: $primary-color;
  }
  &--black {
    color: $color-black;
  }
  &--grey {
    color: $color-grey-dark;
  }
}

/* Custom icons */
.icon-change-adress:before {
  content: '\e900';
}
.icon-refund-delivery-charge:before {
  content: '\e914';
}
.icon-leader:before {
  content: '\e912';
}
.icon-url:before {
  content: '\e913';
}
.icon-currency:before {
  content: '\e90f';
}
.icon-to-left:before {
  content: '\e910';
}
.icon-to-right:before {
  content: '\e911';
}
.icon-address-lookup:before {
  content: '\e90e';
}
.icon-returns:before {
  content: '\e90d';
}
.icon-collected:before {
  content: '\e901';
}
.icon-created:before {
  content: '\e902';
}
.icon-delivered:before {
  content: '\e903';
}
.icon-in-progress:before {
  content: '\e904';
}
.icon-in-transit:before {
  content: '\e905';
}
.icon-ready-to-collect:before {
  content: '\e906';
}
.icon-refunded:before {
  content: '\e907';
}
.icon-return-failed:before {
  content: '\e908';
}
.icon-return-in-progress:before {
  content: '\e909';
}
.icon-returned:before {
  content: '\e90a';
}
.icon-rebook-delivery:before {
  content: '\e90b';
}
.icon-price-history:before {
  content: '\e90c';
}
.icon-pointer:before {
  content: '\e915';
}
.icon-contact-history:before {
  content: '\e916';
}
.icon-order-history:before {
  content: '\e917';
}
.icon-failed-payment:before {
  content: '\e918';
}
.icon-sticky-note:before {
  content: '\e919';
}
