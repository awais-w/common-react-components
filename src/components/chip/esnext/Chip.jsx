import React from 'react';
import PropTypes, { oneOf } from 'prop-types';
import Icon from '@cmc/icon/esnext/Icon';
import bem from '@argos/utils/esnext/bem';
import isEmpty from '@cmc/utils/esnext/isEmpty';
import './Chip.scss';

const cn = bem({ prefix: 'cmc-', block: 'chip' });

const Chip = ({ icon, label, onClick, color }) => (
  <div className={cn(null, color)} onClick={onClick}>
    {!isEmpty(icon) && <Icon isCustom={icon.isCustom} icon={icon.name} />}
    {!isEmpty(label) && <span className={cn('label')}>{label}</span>}
  </div>
);

Chip.propTypes = {
  icon: PropTypes.shape({
    name: PropTypes.string,
    isCustom: PropTypes.bool,
  }).isRequired,
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  onClick: PropTypes.func.isRequired,
  color: oneOf(['amber', 'blue', 'red', 'green', 'grey', 'black']),
};
export default Chip;
