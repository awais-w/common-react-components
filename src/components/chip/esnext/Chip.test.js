import React from 'react';
import { shallow } from 'enzyme';
import Icon from '@cmc/icon/esnext/Icon';
import Chip from './Chip';

const setup = (props) => {
  const component = shallow(<Chip {...props} />);

  return {
    wrapper: component,
    getChip: () => component.find('.cmc-chip'),
    getIcon: () => component.find(Icon),
    getLabel: () => component.find('.cmc-chip__label'),
    getOnClick: () => component.find('.cmc-chip').prop('onClick'),
  };
};

describe('Chip', () => {
  describe('with props', () => {
    const component = setup({
      icon: {
        name: 'icon-name',
        isCustom: false,
      },
      label: 'Chip label',
      color: 'red',
    });

    it('should match default snapshot', () => {
      expect(component.wrapper).toMatchSnapshot();
    });

    it('should render the chip component', () => {
      expect(component.getChip()).toExist();
    });

    it('should render correct color class', () => {
      expect(component.getChip().hasClass('cmc-chip--red')).toBeTruthy();
    });

    it('should render an icon with correct icon name', () => {
      expect(component.getIcon()).toExist();
      expect(component.getIcon().props().icon).toBe('icon-name');
    });

    it('should render a label', () => {
      expect(component.getLabel()).toExist();
      expect(component.getLabel().text()).toBe('Chip label');
    });
  });

  describe('with custom icon', () => {
    const component = setup({
      icon: {
        name: 'custom-icon',
        isCustom: true,
      },
    });

    it('should render a custom icon with correct name', () => {
      expect(component.getChip().find(Icon).props().isCustom).toBe(true);
      expect(component.getChip().find(Icon).props().icon).toBe('custom-icon');
    });
  });

  describe('with onClick prop', () => {
    const onClickProp = jest.fn();
    const component = setup({ onClick: onClickProp });

    it('should have an onClick function', () => {
      expect(component.getOnClick()).toEqual(onClickProp);
    });
  });
});
