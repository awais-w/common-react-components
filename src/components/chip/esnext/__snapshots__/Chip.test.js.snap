// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip with props should match default snapshot 1`] = `
<div
  className="cmc-chip cmc-chip--red"
>
  <Icon
    icon="icon-name"
    isCustom={false}
    onClick={null}
  />
  <span
    className="cmc-chip__label"
  >
    Chip label
  </span>
</div>
`;
